Modules/_queuemodule.c

```diff
@@ -381,55 +381,56 @@ PyDoc_STRVAR(queue_module_doc,
 "C implementation of the Python queue module.\n\
 This module is an implementation detail, please do not use it directly.");
 
-static struct PyModuleDef queuemodule = {
-    .m_base = PyModuleDef_HEAD_INIT,
-    .m_name = "_queue",
-    .m_doc = queue_module_doc,
-    .m_size = sizeof(simplequeue_state),
-    .m_traverse = queue_traverse,
-    .m_clear = queue_clear,
-    .m_free = queue_free,
-};
-
-
-PyMODINIT_FUNC
-PyInit__queue(void)
+static int
+queuemodule_exec(PyObject *module)
 {
-    PyObject *m;
-    simplequeue_state *state;
-
-    /* Create the module */
-    m = PyModule_Create(&queuemodule);
-    if (m == NULL)
-        return NULL;
-
-    state = simplequeue_get_state(m);
+    simplequeue_state *state = simplequeue_get_state(module);
     state->EmptyError = PyErr_NewExceptionWithDoc(
         "_queue.Empty",
         "Exception raised by Queue.get(block=0)/get_nowait().",
         NULL, NULL);
-    if (state->EmptyError == NULL)
-        goto error;
+    if (state->EmptyError == NULL) {
+        return -1;
+    }
 
     Py_INCREF(state->EmptyError);
-    if (PyModule_AddObject(m, "Empty", state->EmptyError) < 0) {
-        Py_DECREF(state->EmptyError);
-        goto error;
+    if (PyModule_AddObject(module, "Empty", state->EmptyError) < 0) {
```

Comment:
I would prefer PyModule_AddObjectRef() here and avoid the INCREF/DECREF dance.

I don't think that there is need to explicitly call Py_CLEAR(state->EmptyError); since the module state will be cleared anyway if the function fails.